### PR TITLE
Symintify repeat_interleave.self_int

### PR DESCRIPTION
**Pull request by SherlockNoMad**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89111

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: torch/fx/graph.py [hunk 1]
```diff
@@ -499,11 +499,11 @@ def emit_node(node : Node):
                 meta_val = node.meta.get('val', node.meta.get('tensor_meta', None))
 
                 if isinstance(meta_val, FakeTensor):
-                    maybe_type_annotation = f': {dtype_abbrs[meta_val.dtype]}{stringify_shape(meta_val.shape)}'
+                    maybe_type_annotation = f': {dtype_abbrs[meta_val.dtype]}{stringify_shape(meta_val.shape)}, {stringify_shape(meta_val.stride())}'
```
Comment (SherlockNoMad) [inline torch/fx/graph.py:502]: I would suggest not to include stride by default.

File: torch/_prims/__init__.py [hunk 2]
```diff
@@ -2323,10 +2323,12 @@ def _arange_meta(
...
         lambda: "step must be nonzero",
     )
-    utils.check(
-        math.isfinite(start) and math.isfinite(end),
-        lambda: f"unsupported range: {start} -> {end}",
-    )
+    # SymInts can't represent inf
+    if not isinstance(start, torch.SymInt) and not isinstance(end, torch.SymInt):
+        utils.check(
+            math.isfinite(start) and math.isfinite(end),
```
Comment (SherlockNoMad) [inline torch/_prims/__init__.py:2329]: how to add sym_isfinite?

Comment (ezyang) [reply torch/_prims/__init__.py:2329]: `sym_isfinite` is pretty easy for int; we just test if is SymInt and return True unconditionally. For float need to actually translate into some sort of condition, but probably doesn't matter too much.

File: torch/fx/graph.py (hunk 1 above)
Comment (ezyang) [reply torch/fx/graph.py:502]: yeah you probably shouldn't put this in this PR

**Review by ezyang — APPROVED** [review]

**Comment by ezyang** [comment]
This is a very incomplete implementation, but the more complete implementation requires adding some new functions (this is the `int_tensor` stuff I discussed with @anjali411 )

**Pull request closed without merging**